docs/proposals/20220215-obsctl.md

```diff
@@ -0,0 +1,67 @@
+# obsctl: A CLI to interact with Observatorium
+
+* **Owners:**
+  * `@onprem` `@saswatamcode`
+
+* **Related Tickets:**
+  * https://issues.redhat.com/browse/MON-2199
+
+* **Other docs:**
+  * None
+
+## Why
+
+We can interact with [Observatorium API](https://github.com/observatorium/api) with a convenient and dedicated CLI, which provides capabilities to persist auth data, and allows us to easily & quickly get up and running with our operations.
+
+## Pitfalls of Current Solution
+
+Currently, we need to manually run [token-refresher](https://github.com/observatorium/token-refresher) with [OIDC](https://openid.net/connect/) information for a particular tenant, and then make cURL requests through that. This is not optimal and generally confusing/trickier for users, both old and new.
+
+## Goals
+
+obsctl will,
+* Manage authentication configuration for multiple tenants by saving them locally
+* Allow users to switch between tenants conveniently
+* Configure and view rules for a tenant
+* View series, rules, and labels for a tenant
+* Later on, support more such one-off operations for other domains as well (logs, traces, alert-routing configs)
+
+## Non-Goals
+
+* Adding and dynamically configuring tenants through obsctl (No such admin API is available within Observatorium, nor is it planned)
+
+## Audience of this proposal
+
+Observatorium Devs and Users.
+
+## How
+
+As obsctl is a CLI, authentication options (OIDC) need to be persisted across sessions. obsctl aims to achieve this by saving those configurations locally at some configuration directory and maintaining a "current" tenant.
+
+The "current" tenant will basically represent the logged-in tenant and all operations will be done based on this "current" tenant configuration. In case, the "current" tenant is empty, i.e, there is no tenant currently logged-in, obsctl will error out.
```

Comment:
How current tenant will work for admin sessions?

Reply:
When we introduce admin level APIs in Observatroium, we still would need to have authentication on top of it, maybe we can use the same context to store admin credentials as well. The whole picture around how admin endpoints will work is not clear as of now so I think we can think about this in future.